Selinux facts

bcoca opened the pull request:
##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
more accureate reporting on selinux

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - Bugfix Pull Request


##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
facts

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
2.4
```


##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
  -->
alternatively fixes #13306
fixes https://github.com/ansible/ansible-modules-core/issues/2432
<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

mattclay commented:
CI failure in integration tests. Here's one example:

```
2017-05-17 15:42:15 TASK [sefcontext : debug] ******************************************************
2017-05-17 15:42:15 fatal: [testhost]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'dict object' has no attribute 'status'\n\nThe error appears to have been in '/root/ansible/test/integration/targets/sefcontext/tasks/main.yml': line 23, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- debug:\n  ^ here\n"}
```

bcoca committed:
added new selinux fact to clarify python lib

the selinux fact is boolean false when the library is not installed,
a dictionary/hash otherwise, but this is ambigous
added new fact so we can eventually remove the type dichtomy and normalize it as a dict

defionscode commented:
shipit
